core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/speech/recognizer/RecognizerCreator.java

```diff
@@ -0,0 +1,180 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information see notice.md
+ */
+
+package de.unistuttgart.iaas.amyassist.amy.core.speech.recognizer;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.NoSuchElementException;
+import java.util.Properties;
+
+import javax.sound.sampled.AudioInputStream;
+
+import org.slf4j.Logger;
+
+import de.unistuttgart.iaas.amyassist.amy.core.audio.AudioManager;
+import de.unistuttgart.iaas.amyassist.amy.core.audio.LocalAudio;
+import de.unistuttgart.iaas.amyassist.amy.core.configuration.ConfigurationManager;
+import de.unistuttgart.iaas.amyassist.amy.core.di.annotation.Reference;
+import de.unistuttgart.iaas.amyassist.amy.core.di.annotation.Service;
+import de.unistuttgart.iaas.amyassist.amy.core.io.Environment;
+import de.unistuttgart.iaas.amyassist.amy.core.natlang.NLProcessingManager;
+import de.unistuttgart.iaas.amyassist.amy.core.service.DeploymentContainerService;
+import de.unistuttgart.iaas.amyassist.amy.remotesr.RemoteSR;
+
+/**
+ * Class that Creates all Grammar Objects sets the MainGrammar and the List of all other Grammars
+ * 
+ * @author Kai Menzel
+ */
+
+@Service(RecognizerCreator.class)
+public class RecognizerCreator implements DeploymentContainerService {
+
+	// --------------------------------------------------------------
+	// Dependencies
+
+	@Reference
+	private Logger logger;
+	@Reference
+	private Environment environment;
+	@Reference
+	private NLProcessingManager nlProcessingManager;
+	@Reference
+	private RemoteSR googleRecognition;
+	@Reference
+	private AudioManager audioManager;
+	@Reference
+	private LocalAudio localAudio;
+	@Reference
+	private ConfigurationManager configManager;
+
+	// --------------------------------------------------------------
+	// Fields
+
+	private Properties config;
+
+	private static final String CONFIG_NAME = "localSpeech.config";
+	private static final String PROPERTY_ENABLE = "enable";
+
+	private boolean recognitionDisabled = false;
+
+	private String mainGrammarName = "mainGrammar";
+	// private String tempGrammarName = "tempGrammar"
+
+	private AudioInputStream ais = null;
+
+	// --------------------------------------------------------------
+	// Construcor
+
+	// --------------------------------------------------------------
+	// Init
+
+	/**
+	 * Call this after all register and before process
```

Comment:
What does this comment try to tell me? The time this is called is defined by the Interface. If no further documentation about the effect of the method is needed, remove this comment.